networkx/algorithms/shortest_paths/astar.py

```diff
@@ -9,6 +9,7 @@
 __all__ = ["astar_path", "astar_path_length"]
 
 
+@nx._dispatch(edge_attrs="weight", preserve_node_attrs="heuristic")
```

Comment:
`preserve_node_attrs="heuristic"` means _all_ node attributes are preserved when the `heuristic` argument is a callable.